controllers/serviceexport/unit_test.go

```diff
@@ -0,0 +1,31 @@
+package serviceexport_test
+
+import (
+	"testing"
+	"time"
+
+	"go.goms.io/fleet-networking/controllers/serviceexport"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	fakedynamic "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/informers"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// TestNewController calls serviceexport.New() with fake Kubernetes clients, checking for an error.
+func TestNewController(t *testing.T) {
+	memberKubeClient := fakeclientset.NewSimpleClientset()
+	memberDynamicClient := fakedynamic.NewSimpleDynamicClient(scheme.Scheme)
+	hubDynamicClient := fakedynamic.NewSimpleDynamicClient(scheme.Scheme)
+
+	memberSharedInformerFactory := informers.NewSharedInformerFactory(memberKubeClient, time.Second*30)
+	memberDynamicInformerFactory := dynamicinformer.NewDynamicSharedInformerFactory(memberDynamicClient, time.Second*30)
+	hubDynamicInformerFactory := dynamicinformer.NewDynamicSharedInformerFactory(hubDynamicClient, time.Second*30)
+
+	_, err := serviceexport.New("fake-cluster", memberKubeClient, memberDynamicClient, hubDynamicClient,
```

Comment:
can we compare the outputs if possible?

Reply:
Ah... The only comparable field would be the cluster ID and the passed-in clients. It would be a relatively weird comparison by my standard TBH; though adding them would not hurt I guess.